### PR TITLE
Detection of already installed homebrew cask

### LINKED ISSUE #864: Homebrew module incorrectly fails executing task when package is already installed
##### SUMMARY
```
- name: Install wkhtmltopdf
  homebrew:
    name: homebrew/cask/wkhtmltopdf 
    state: present
```

results in

```
PLAY [localhost] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [Install wkhtmltopdf] *****************************************************
fatal: [localhost]: FAILED! => changed=false 
  msg: |-
    Updating Homebrew...
    Warning: Cask 'wkhtmltopdf' is already installed.
  
    To re-install wkhtmltopdf, run:
      brew cask reinstall wkhtmltopdf

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
[Homebrew module](https://docs.ansible.com/ansible/latest/modules/homebrew_module.html)

##### ANSIBLE VERSION

```paste below
$ ansible --version
ansible 2.9.13
  config file = /Users/mpereira/git/macbook-playbook/ansible.cfg
  configured module search path = ['/Users/mpereira/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mpereira/Library/Python/3.8/lib/python/site-packages/ansible
  executable location = /Users/mpereira/Library/Python/3.8/bin/ansible
  python version = 3.8.5 (v3.8.5:580fbb018f, Jul 20 2020, 12:11:27) [Clang 6.0 (clang-600.0.57)]
```

##### CONFIGURATION

```paste below
$ ansible-config dump --only-changed
DEFAULT_STDOUT_CALLBACK(/Users/mpereira/git/macbook-playbook/ansible.cfg) = yaml
```

##### OS / ENVIRONMENT

```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.15.5
BuildVersion:	19F101
```

##### STEPS TO REPRODUCE

Refer to _summary_.

##### EXPECTED RESULTS

I expect the task to complete successfully when a package is already installed and a `homebrew` task is executed with either `state: present` or `state: latest`.

##### ACTUAL RESULTS

The task fails, as shown in the _summary_.

### PR DESCRIPTION
**Pull request by joao-victor-silva**
Use json output v2 to check if formulae and casks are installed

##### SUMMARY

Fixes #864 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

modules/homebrew

##### ADDITIONAL INFORMATION

The current method to check if a cask or formulae is installed tries to detect a string (`Built from source` or `Poured from bottle`) piece in the output, but for casks the strings aren't present anymore (example below). The suggested solution for it is use the json output of the command to check the `installed` field that contains a list of installed versions for formulae or a installed string version for casks.

`brew info firefox` output
```
==> firefox: 121.0.1 (auto_updates)
https://www.mozilla.org/firefox/
/usr/local/Caskroom/firefox/121.0.1 (226B)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/f/firefox.rb
==> Name
Mozilla Firefox
==> Description
Web browser
==> Languages
af, ar, be, bg, bn, ca, cs, de, en-CA, en-GB, en, eo, es-AR, es-CL, es-ES, fa, ff, fi, fr, gl, gn, gu, he, hi, in, it, ja, ka, ko, mr, my, ne, nl, pa-IN, pl, pt-BR, pt, ru, si, sq, sr, sv, ta, te, th, tl, tr, uk, ur, zh-TW, zh
==> Artifacts
Firefox.app (App)
/usr/local/Caskroom/firefox/121.0.1/firefox.wrapper.sh -> firefox (Binary)
==> Analytics
install: 11,427 (30 days), 34,823 (90 days), 118,245 (365 days)
```

`brew info --json=v2 firefox | jq '.casks[0].installed'` output (with jq for brevity)
```json
"121.0.1"
```

### CODE REVIEW

File: changelogs/fragments/7870-homebrew-cask-installed-detection.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - homebrew - detect already installed formulae and casks using json output from brew info (https://github.com/ansible-collections/community.general/issues/864).
```
Comment (felixfontein) [inline changelogs/fragments/7870-homebrew-cask-installed-detection.yml:2]:
```suggestion
  - homebrew - detect already installed formulae and casks using JSON output from ``brew info`` (https://github.com/ansible-collections/community.general/issues/864).
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: changelogs/fragments/7870-homebrew-cask-installed-detection.yml (hunk 1 above)
Comment (joao-victor-silva) [reply changelogs/fragments/7870-homebrew-cask-installed-detection.yml]: Thank you, I'd applied the suggestions and pushed the new version.

**Review by felixfontein — COMMENTED** [review]
If nobody objects, I'll merge this the next weekend!

**Comment by felixfontein** [comment]
@joao-victor-silva thanks a lot for your contribution!

**Comment by joao-victor-silva** [comment]
Thanks @felixfontein, I really appreciate all the help in this PR ❤️ .

**Pull request merged**